compiler/rustc_mir_build/src/builder/matches/buckets.rs

```diff
@@ -291,7 +289,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                     if !test.overlaps(pat, self.tcx)? { Some(TestBranch::Failure) } else { None }
                 }
             }
-            (TestKind::Range(range), &TestableCase::Constant { value }) => {
+            (TestKind::Range(range), &TestableCase::Constant { value, kind: _ }) => {
```

Comment:
I'd be happy to merge if you add `PatConstKind::Float` and change this line:
```suggestion
            (TestKind::Range(range), &TestableCase::Constant { value, kind: Bool | IntOrChar | Float }) => {
```
Otherwise I feel like this is slightly adding confusion on top of the existing one.